client/src/components/BlogPostForm.tsx

```diff
@@ -19,100 +20,78 @@ type props = {
 export default function BlogPostForm({ entry }: props) {
   const navigate = useNavigate();
   const { user, token } = useUser();
-  const [title, setTitle] = useState(entry?.title || '');
-  const [subtitle, setSubtitle] = useState(entry?.subtitle || '');
-  const [location, setLocation] = useState(entry?.location || '');
-  const [body, setBody] = useState(entry?.body || '');
-  const [entryId] = useState(entry?.entryId);
   const [search, setSearch] = useState('');
   const [photos, setPhotos] = useState<Photo[]>();
-  const [missingPhoto, setMissingPhoto] = useState(entry ? false : undefined);
+  const [missingPhoto, setMissingPhoto] = useState(false);
   const [loadingAnimation, setLoadingAnimation] = useState(false);
+  const [loading, setLoading] = useState(false);
   const [error, setError] = useState<unknown>();
-  const [photoInfo, setPhotoInfo] = useState<Photo | undefined>(
-    entry
-      ? {
-          id: 'original',
-          urls: {
-            full: entry.photoURLBig,
-            regular: entry.photoURL,
-          },
-          user: {
-            name: entry.photoAuthor,
-            links: {
-              html: entry.photoAuthorLink,
-            },
-          },
-          alt_description: entry.photoAlt,
-        }
-      : undefined
+  const [fullEntry, setFullEntry] = useState(
```

Comment:
This is a fairly common approach to forms with many fields. It makes a lot of the code simpler.  I wonder if it would be even simpler if you went with uncontrolled form fields, but I'm not sure it's worth spending the time on.

If you were to use the popular React Hook Form package for managing your forms, it would follow an approach that is very similar to what you've done now. You would have a `fullEntry` value, but instead of it being in your state, you would pass it as the default value to your form hook, which would keep it in its state. Your form fields would almost all be uncontrolled, as well.

So, what you've done here is a good approach.